optaplanner-examples/src/main/java/org/optaplanner/examples/flightcrewscheduling/persistence/FlightCrewSchedulingGenerator.java

```diff
@@ -0,0 +1,298 @@
+                        END_MINUTE_OF_DAY - flyingTime - START_MINUTE_OF_DAY + 1);
+                int arrivalMinute = departureMinute + flyingTime;
+                for (LocalDate date = firstDate; date.compareTo(lastDate) <= 0; date = date.plusDays(1)) {
+                    Flight flight = new Flight();
+                    flight.setId((long) flightId);
+                    flightId++;
+                    flight.setFlightNumber(flightNumber);
+                    flight.setDepartureAirport(departureAirport);
+                    flight.setDepartureUTCDateTime(date.atTime(departureMinute / 60, departureMinute % 60));
+                    flight.setArrivalAirport(arrivalAirport);
+                    flight.setArrivalUTCDateTime(date.atTime(arrivalMinute / 60, arrivalMinute % 60));
+                    for (int indexInFlight = 0; indexInFlight < EMPLOYEE_COUNT_PER_FLIGHT; indexInFlight++) {
+                        FlightAssignment flightAssignment = new FlightAssignment();
+                        flightAssignment.setId((long) flightAssignmentId);
+                        flightAssignmentId++;
+                        flightAssignment.setFlight(flight);
+                        flightAssignment.setIndexInFlight(indexInFlight);
+                        Skill requiredSkill = indexInFlight < PILOT_COUNT_PER_FLIGHT ? pilotSkill : flightAttendantSkill;
+                        flightAssignment.setRequiredSkill(requiredSkill);
+                        flightAssignmentList.add(flightAssignment);
+                    }
+                    logger.trace("Created flight ({}).", flight);
+                    flightList.add(flight);
+                }
+            }
+        }
+        flightList.sort(Comparator.comparing(Flight::getDepartureUTCDate)
+                .thenComparing(Flight::getDepartureAirport)
+                .thenComparing(Flight::getDepartureUTCDateTime)
+                .thenComparing(Flight::getArrivalAirport)
+                .thenComparing(Flight::getArrivalUTCDateTime));
+        solution.setFlightList(flightList);
+        solution.setFlightAssignmentList(flightAssignmentList);
+    }
+
+
+    private void createEmployeeList(FlightCrewSolution solution, int flightRoundTripsPerDay, int dayCount) {
+        int employeeListSize = flightRoundTripsPerDay * EMPLOYEE_COUNT_PER_FLIGHT * 3;
+        List<Employee> employeeList = new ArrayList<>(employeeListSize);
+        employeeNameGenerator.predictMaximumSizeAndReset(employeeListSize);
+        LocalDate firstDate = solution.getScheduleFirstUTCDate();
+        LocalDate lastDate = solution.getScheduleLastUTCDate();
+        List<LocalDate> allDayList = new ArrayList<>((int) DAYS.between(firstDate, lastDate) + 1);
+        for (LocalDate date = firstDate; date.compareTo(lastDate) <= 0; date = date.plusDays(1)) {
+            allDayList.add(date);
+        }
+        List<LocalDate> unavailableDayPool = new ArrayList<>(allDayList);
```

Comment:
You have a good point,
but due to the flightAssignmentId and the flightId (which need to be returned from methods) this impacts readability too.
In general, I am more lenient on method length and nesting dept for Generator and FileIO code than I am for other code, because it's really just handling each property one by one.

Let me see if I can split off the FlightAssignment creation entirely, instead of just extracting method of that 4th for loop.